src/main/backend.ts

```diff
@@ -150,30 +142,95 @@ export class MainBackend extends BaseBackend<ElectronOptions> implements CommonB
-        const cloned = Scope.clone(updatedScope);
+        const scope = Scope.clone(updatedScope);
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._eventProcessors = [];
+        (scope as any)._eventProcessors = [];
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (cloned as any)._scopeListeners = [];
+        (scope as any)._scopeListeners = [];
+
         // if we use the crashpad minidump uploader we have to set extra whenever the scope updates
         if (this._options.useCrashpadMinidumpUploader === true) {
-          getCurrentHub().captureEvent(
-            {
-              // @ts-ignore __INTERNAL_MINIDUMP is not assignable to event
-              __INTERNAL_MINIDUMP: true,
-            },
-            {
-              data: {
-                __sentry__: true,
-              },
-            },
+          forget(
+            this._getNativeUploaderEvent(scope).then(event => {
+              // Update the extra parameters in the main process
+              const mainParams = this._getNativeUploaderExtraParams(event);
+              for (const key of Object.keys(mainParams)) {
+                crashReporter.addExtraParameter(key, mainParams[key]);
+              }
+
+              // Modify the event to point to renderer crash
+              event.extra = { ...event.extra, crashed_process: 'renderer' };
+              // Update the extra parameters in every renderer process
+              const rendererParams = this._getNativeUploaderExtraParams(event);
+              for (const window of BrowserWindow.getAllWindows()) {
+                if (!window.webContents.isDestroyed()) {
+                  window.webContents.send(IPC.EXTRA_PARAMS, rendererParams);
+                }
+              }
+            }),
           );
         }
-        this._scopeStore.set(cloned);
+
+        this._scopeStore.set(scope);
       });
     }
   }
 
+  /** Builds up an event to send with the native Electron uploader */
+  private async _getNativeUploaderEvent(scope: Scope): Promise<Event> {
+    // Basic SDK info
+    let event: Event = {
+      tags: { event_type: 'native' },
+    };
+
+    // Apply the scope to the event
+    await scope.applyToEvent(event);
+    // Add all the extra context
+    event = await addEventDefaults(this._options.appName, event);
+    return normalizeEvent(event);
+  }
+
+  /** Chunks up event JSON into 1 or more parameters for use with the native Electron uploader
+   *
+   * Returns chunks with keys and values:
+   * {
+   *    sentry__1: '{ json...',
+   *    sentry__2: 'more json...',
+   *    sentry__x: 'end json }',
+   * }
+   */
+  private _getNativeUploaderExtraParams(event: Event): { [key: string]: string } {
+    // We chunk on Linux too because the built in chunking is limited to 160 * 127 byte parameters
+    const maxBytes = process.platform === 'linux' ? 120 : 20300;
```

Comment:
This will change once Electron uses Crashpad on Linux. Let's keep this in mind so we can introduce an Electron version switch in the future.